File: 733-flood-fill/733-flood-fill.cpp
#include <queue>
class Solution {
public:
    vector<vector<int>> floodFill(vector<vector<int>>& image, int x, int y, int newColor) {
        int oldColor  = image[x][y],i,j ;
        int m = image.size(),n=image[0].size();
        queue<pair<int,int>> q;
        q.push({x,y});
        while(!q.empty()){
            auto pixel = q.front();
            i = pixel.first;
            j = pixel.second;
            q.pop();
            
            if( i<m && j<n && i>=0 && j>=0){
                if (image[i][j] == oldColor && image[i][j] != newColor)
                {
                image[i][j]=newColor;
                q.push({i+1,j});
                q.push({i-1,j});
                q.push({i,j+1});
                q.push({i,j-1});
                }
            }
        }
        return image;
}
        
    
};

File: 733-flood-fill/733-flood-fill.py
from collections import deque

class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        
        h,w=len(image),len(image[0])
        
        def dfs(cur_r,cur_c,src_color,new_color):
                        
            if cur_r<0 or cur_r>=h or cur_c<0 or cur_c>=w or image[cur_r][cur_c]==newColor or image[cur_r][cur_c]!=src_color:
                return
            
            # update color
            image[cur_r][cur_c]=newColor
            
            #BFS to 4 connected neighbours
            dfs(cur_r-1,cur_c,src_color,new_color) 
            dfs(cur_r+1,cur_c,src_color,new_color) 
            dfs(cur_r,cur_c-1,src_color,new_color)
            dfs(cur_r,cur_c+1,src_color,new_color) 
        dfs(sr,sc,src_color=image[sr][sc],new_color = newColor) 
        return image



        
        